### ISSUE TITLE
0.8.9 changes exit status behavior in a breaking way

### ISSUE THREAD

**Issue by Iristyle** [derailed/popeye#122]
<img src="https://raw.githubusercontent.com/derailed/popeye/master/assets/popeye_feature.png" align="right" width="100" height="auto"/>

<br/>
<br/>
<br/>

**Is your feature request related to a problem? Please describe.**

In 0.8.9, https://github.com/derailed/popeye/commit/94da2df5e9ecdabd7565222720cea43a285bcc01 changed the behavior to produce a non-zero exit status with the presence of warnings or errors.

This unfortunately broke our existing CI workflow that did the following:

```
	docker run --rm -i -v ${HOME}/${CONFIG_PATH}:/root/.kube ${DOCKER_RUN_SWITCHES} quay.io/derailed/popeye:latest --namespace ${KOTS_NAMESPACE} --lint info
	docker run --rm -i -v ${HOME}/${CONFIG_PATH}:/root/.kube ${DOCKER_RUN_SWITCHES} quay.io/derailed/popeye:latest --namespace ${KOTS_NAMESPACE} --lint info --out json | jq '[.popeye.sanitizers[].tally.error] | any (. >= 1) | not' --exit-status
```

We ran popeye once for CI / informational purposes... and then again to figure out the error count by parsing JSON. We were ignoring warnings.

Now with warnings producing non-zero exit, only the first line runs and CI dies, when it previously continued.

**Describe the solution you'd like**

There are a couple of things I'd like to see if possible:

* A cli switch to opt into treating warnings as errors (IMHO, warnings shouldn't generate non-zero exit codes)
* Related, it would be great to be able to produce multiple outputs in a single run (i.e. report for CI + the json file on disk)

**Describe alternatives you've considered**

I should be able to work around this new issue with ` || true` on the first line, but a cli switch would be better self-documenting.

Thanks!

**Comment by derailed** [comment]
@Iristyle Rats! Sorry I've broke your build ;( I was debating about that and I think you're right in terms on the non zero exit only on error. I'll fix in the next drop. As far as the dual output I think the CI should just monitor the exit status (provided error only) and you can get further detail in the actual reports. Would that work for you?

**Comment by premkumar-gg** [comment]
Has this been fixed yet? If we run popeye as a CronJob in a cluster, the pod will end up in error state, leaving it to restart a number of times.

**Comment by premkumar-gg** [comment]
Gotcha.. I understand why a non-zero exit is necessary, and I see the reported issue here had been fixed in 0.8.10 👍 

But, it seems we cannot run popeye as a CronJob anymore as it returns a non-zero exit.
